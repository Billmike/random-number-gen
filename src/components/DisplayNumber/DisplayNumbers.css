.number-container {
  align-self: center;
  margin-top: 20px;
  max-height: 400px;
  overflow-y: scroll;
}

.number-count {
  margin-bottom: 30px
}

.phone-number {
  margin-bottom: 15px;
}