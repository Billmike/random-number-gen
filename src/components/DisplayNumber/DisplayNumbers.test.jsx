import React from 'react';
import { shallow } from 'enzyme';
import DisplayNumbers from './index';

test('<DisplayNumbers /> mounts correctly', () => {
  const props = {
    phoneNumbers: [
      {userPhoneNumber: 123456},
      {userPhoneNumber: 123456},
      {userPhoneNumber: 123456},
      {userPhoneNumber: 123456},
      {userPhoneNumber: 123456},
      {userPhoneNumber: 123456},
      {userPhoneNumber: 123456},
      {userPhoneNumber: 123456},
      {userPhoneNumber: 123456},
      {userPhoneNumber: 123456},
      {userPhoneNumber: 123456},
      {userPhoneNumber: 123456},
      {userPhoneNumber: 123456},
      {userPhoneNumber: 123456},
      {userPhoneNumber: 123456},
      {userPhoneNumber: 123456}
    ]
  };

  const wrapper = shallow(<DisplayNumbers {...props} />);

  expect(wrapper).toMatchSnapshot()
})

test('should verify the handlePageClick method is called correctly', () => {
  const props = {
    phoneNumbers: [
      {userPhoneNumber: 123456},
      {userPhoneNumber: 123456},
      {userPhoneNumber: 123456},
      {userPhoneNumber: 123456},
      {userPhoneNumber: 123456},
      {userPhoneNumber: 123456},
      {userPhoneNumber: 123456},
      {userPhoneNumber: 123456},
      {userPhoneNumber: 123456},
      {userPhoneNumber: 123456},
      {userPhoneNumber: 123456},
      {userPhoneNumber: 123456},
      {userPhoneNumber: 123456},
      {userPhoneNumber: 123456},
      {userPhoneNumber: 123456},
      {userPhoneNumber: 123456}
    ]
  };
  const phoneNumbers = [
    {userPhoneNumber: 123456},
      {userPhoneNumber: 123456},
      {userPhoneNumber: 123456},
      {userPhoneNumber: 123456},
      {userPhoneNumber: 123456},
      {userPhoneNumber: 123456},
      {userPhoneNumber: 123456},
      {userPhoneNumber: 123456},
      {userPhoneNumber: 123456},
      {userPhoneNumber: 123456},
      {userPhoneNumber: 123456},
      {userPhoneNumber: 123456},
      {userPhoneNumber: 123456},
      {userPhoneNumber: 123456},
      {userPhoneNumber: 123456},
      {userPhoneNumber: 123456}
  ]
  const wrapper = shallow(<DisplayNumbers {...props} />);
  const instance = wrapper.instance();
  wrapper.setState({ generatedPhoneNumbers: phoneNumbers });
  const data = {
    selected: ''
  }
  instance.handlePageClick(data);
  expect(wrapper.state().generatedPhoneNumbers).toEqual(phoneNumbers)
});

test('should test get derived state from props', () => {
  const props = {
    phoneNumbers: [
      {userPhoneNumber: 123456},
      {userPhoneNumber: 123456},
      {userPhoneNumber: 123456},
      {userPhoneNumber: 123456},
      {userPhoneNumber: 123456},
      {userPhoneNumber: 123456},
      {userPhoneNumber: 123456},
    ],
    hasGeneratedNumbers: true
  };
  const state = {
    currentPage: 1
  };

  const wrapper = shallow(<DisplayNumbers {...props} />);
  const result = DisplayNumbers.getDerivedStateFromProps(props, state);
})

