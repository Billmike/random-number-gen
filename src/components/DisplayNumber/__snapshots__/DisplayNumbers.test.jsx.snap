// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DisplayNumbers /> mounts correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <DisplayNumber
    phoneNumbers={
      Array [
        Object {
          "userPhoneNumber": 123456,
        },
        Object {
          "userPhoneNumber": 123456,
        },
        Object {
          "userPhoneNumber": 123456,
        },
        Object {
          "userPhoneNumber": 123456,
        },
        Object {
          "userPhoneNumber": 123456,
        },
        Object {
          "userPhoneNumber": 123456,
        },
        Object {
          "userPhoneNumber": 123456,
        },
        Object {
          "userPhoneNumber": 123456,
        },
        Object {
          "userPhoneNumber": 123456,
        },
        Object {
          "userPhoneNumber": 123456,
        },
        Object {
          "userPhoneNumber": 123456,
        },
        Object {
          "userPhoneNumber": 123456,
        },
        Object {
          "userPhoneNumber": 123456,
        },
        Object {
          "userPhoneNumber": 123456,
        },
        Object {
          "userPhoneNumber": 123456,
        },
        Object {
          "userPhoneNumber": 123456,
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        Array [
          <div
            className="number-style"
          >
            090123456
          </div>,
          <div
            className="number-style"
          >
            090123456
          </div>,
          <div
            className="number-style"
          >
            090123456
          </div>,
          <div
            className="number-style"
          >
            090123456
          </div>,
          <div
            className="number-style"
          >
            090123456
          </div>,
          <div
            className="number-style"
          >
            090123456
          </div>,
          <div
            className="number-style"
          >
            090123456
          </div>,
          <div
            className="number-style"
          >
            090123456
          </div>,
          <div
            className="number-style"
          >
            090123456
          </div>,
          <div
            className="number-style"
          >
            090123456
          </div>,
        ],
        <Pagination
          handlePageClick={[Function]}
          pages={2}
        />,
      ],
      "className": "container-style",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "host",
        "props": Object {
          "children": "090123456",
          "className": "number-style",
        },
        "ref": null,
        "rendered": "090123456",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "host",
        "props": Object {
          "children": "090123456",
          "className": "number-style",
        },
        "ref": null,
        "rendered": "090123456",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": "2",
        "nodeType": "host",
        "props": Object {
          "children": "090123456",
          "className": "number-style",
        },
        "ref": null,
        "rendered": "090123456",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": "3",
        "nodeType": "host",
        "props": Object {
          "children": "090123456",
          "className": "number-style",
        },
        "ref": null,
        "rendered": "090123456",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": "4",
        "nodeType": "host",
        "props": Object {
          "children": "090123456",
          "className": "number-style",
        },
        "ref": null,
        "rendered": "090123456",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": "5",
        "nodeType": "host",
        "props": Object {
          "children": "090123456",
          "className": "number-style",
        },
        "ref": null,
        "rendered": "090123456",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": "6",
        "nodeType": "host",
        "props": Object {
          "children": "090123456",
          "className": "number-style",
        },
        "ref": null,
        "rendered": "090123456",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": "7",
        "nodeType": "host",
        "props": Object {
          "children": "090123456",
          "className": "number-style",
        },
        "ref": null,
        "rendered": "090123456",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": "8",
        "nodeType": "host",
        "props": Object {
          "children": "090123456",
          "className": "number-style",
        },
        "ref": null,
        "rendered": "090123456",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": "9",
        "nodeType": "host",
        "props": Object {
          "children": "090123456",
          "className": "number-style",
        },
        "ref": null,
        "rendered": "090123456",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "handlePageClick": [Function],
          "pages": 2,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          Array [
            <div
              className="number-style"
            >
              090123456
            </div>,
            <div
              className="number-style"
            >
              090123456
            </div>,
            <div
              className="number-style"
            >
              090123456
            </div>,
            <div
              className="number-style"
            >
              090123456
            </div>,
            <div
              className="number-style"
            >
              090123456
            </div>,
            <div
              className="number-style"
            >
              090123456
            </div>,
            <div
              className="number-style"
            >
              090123456
            </div>,
            <div
              className="number-style"
            >
              090123456
            </div>,
            <div
              className="number-style"
            >
              090123456
            </div>,
            <div
              className="number-style"
            >
              090123456
            </div>,
          ],
          <Pagination
            handlePageClick={[Function]}
            pages={2}
          />,
        ],
        "className": "container-style",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "host",
          "props": Object {
            "children": "090123456",
            "className": "number-style",
          },
          "ref": null,
          "rendered": "090123456",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "host",
          "props": Object {
            "children": "090123456",
            "className": "number-style",
          },
          "ref": null,
          "rendered": "090123456",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "host",
          "props": Object {
            "children": "090123456",
            "className": "number-style",
          },
          "ref": null,
          "rendered": "090123456",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "3",
          "nodeType": "host",
          "props": Object {
            "children": "090123456",
            "className": "number-style",
          },
          "ref": null,
          "rendered": "090123456",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "4",
          "nodeType": "host",
          "props": Object {
            "children": "090123456",
            "className": "number-style",
          },
          "ref": null,
          "rendered": "090123456",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "5",
          "nodeType": "host",
          "props": Object {
            "children": "090123456",
            "className": "number-style",
          },
          "ref": null,
          "rendered": "090123456",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "6",
          "nodeType": "host",
          "props": Object {
            "children": "090123456",
            "className": "number-style",
          },
          "ref": null,
          "rendered": "090123456",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "7",
          "nodeType": "host",
          "props": Object {
            "children": "090123456",
            "className": "number-style",
          },
          "ref": null,
          "rendered": "090123456",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "8",
          "nodeType": "host",
          "props": Object {
            "children": "090123456",
            "className": "number-style",
          },
          "ref": null,
          "rendered": "090123456",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "9",
          "nodeType": "host",
          "props": Object {
            "children": "090123456",
            "className": "number-style",
          },
          "ref": null,
          "rendered": "090123456",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "handlePageClick": [Function],
            "pages": 2,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
